tensorflow/compiler/mlir/tensorflow/ir/tf_ops_a_m.cc

```diff
@@ -2215,6 +2215,19 @@ void ExpandDimsOp::build(OpBuilder &builder, OperationState &result,
   return build(builder, result, InferExpandDimsOpType(input, dim), input, dim);
 }
 
+//===----------------------------------------------------------------------===//
+// Expm1Op
+//===----------------------------------------------------------------------===//
+LogicalResult Expm1Op::inferReturnTypes(
+    MLIRContext* context, Optional<Location> location, ValueRange operands,
+    DictionaryAttr attributes, RegionRange regions,
+    SmallVectorImpl<Type>& inferredReturnTypes) {
+  const TensorType input_ty = operands[0].getType().template cast<TensorType>();
```

Comment:
nit: don't assume that input type is a tensor type. We generally assume in other places as the IR is verified but here we should fail if the input is not a tensor type.